metadata-ingestion/tests/integration/vertexai/test_vertexai_source.py

```diff
@@ -0,0 +1,174 @@
+@patch("google.cloud.aiplatform.AutoMLImageTrainingJob.list")
+@patch("google.cloud.aiplatform.AutoMLVideoTrainingJob.list")
+@patch("google.cloud.aiplatform.AutoMLForecastingTrainingJob.list")
+def test_vertexai_source_ingestion(
+    mock_automl_forecasting_job_list: List[VertexAiResourceNoun],
+    mock_automl_video_job_list: List[VertexAiResourceNoun],
+    mock_automl_image_list: List[VertexAiResourceNoun],
+    mock_automl_text_job_list: List[VertexAiResourceNoun],
+    mock_automl_tabular_job_list: List[VertexAiResourceNoun],
+    mock_custom_python_job_list: List[VertexAiResourceNoun],
+    mock_custom_container_job_list: List[VertexAiResourceNoun],
+    mock_custom_training_job_list: List[VertexAiResourceNoun],
+    mock_custom_job_list: List[VertexAiResourceNoun],
+    mock_video_ds_list: List[VertexAiResourceNoun],
+    mock_time_series_ds_list: List[VertexAiResourceNoun],
+    mock_image_ds_list: List[VertexAiResourceNoun],
+    mock_tabular_ds_list: List[VertexAiResourceNoun],
+    mock_text_ds_list: List[VertexAiResourceNoun],
+    mock_model_list: List[Model],
+    mock_init: MagicMock,
+    pytestconfig: Config,
+    sink_file_path: str,
+    pipeline_config: Dict[str, Any],
+    mock_models: List[Model],
+    mock_training_jobs: List[VertexAiResourceNoun],
+) -> None:
+    assert hasattr(mock_model_list, "return_value")
+    mock_model_list.return_value = mock_models
+    assert hasattr(mock_text_ds_list, "return_value")
+    mock_text_ds_list.return_value = []
+    assert hasattr(mock_tabular_ds_list, "return_value")
+    mock_tabular_ds_list.return_value = []
+    assert hasattr(mock_image_ds_list, "return_value")
+    mock_image_ds_list.return_value = []
+    assert hasattr(mock_time_series_ds_list, "return_value")
+    mock_time_series_ds_list.return_value = []
+    assert hasattr(mock_video_ds_list, "return_value")
+    mock_video_ds_list.return_value = []
+    assert hasattr(mock_custom_job_list, "return_value")
+    mock_custom_job_list.return_value = mock_training_jobs
+    assert hasattr(mock_custom_training_job_list, "return_value")
+    mock_custom_training_job_list.return_value = []
+    assert hasattr(mock_custom_container_job_list, "return_value")
+    mock_custom_container_job_list.return_value = []
+    assert hasattr(mock_custom_python_job_list, "return_value")
+    mock_custom_python_job_list.return_value = []
+    assert hasattr(mock_automl_tabular_job_list, "return_value")
+    mock_automl_tabular_job_list.return_value = []
+    assert hasattr(mock_automl_text_job_list, "return_value")
+    mock_automl_text_job_list.return_value = []
+    assert hasattr(mock_automl_image_list, "return_value")
+    mock_automl_image_list.return_value = []
+    assert hasattr(mock_automl_video_job_list, "return_value")
+    mock_automl_video_job_list.return_value = []
+    assert hasattr(mock_automl_forecasting_job_list, "return_value")
+    mock_automl_forecasting_job_list.return_value = []
```

Comment:
this amount of copy-pasted code does not really smell right - there's better ways to do this in python 

```suggestion
with contextlib.ExitStack() as exit_stack:
	for path_to_mock in ['google.cloud.aiplatform.datasets.TextDataset.list', '...', ...]:
		mock = exit_stack.enter_context(patch(path_to_mock))
		mock.return_value = []
	... # rest of the test here
```

Reply:
good suggestion, fixed.